src/main/java/com/esri/samples/kml/play_a_kml_tour/PlayAKMLTourSample.java

```diff
@@ -0,0 +1,207 @@
+      scene.setBaseSurface(surface);
+
+      // create play/pause button
+      final ImageView playIcon = new ImageView(new Image(PlayAKMLTourSample.class.getResourceAsStream(
+          "/icons/play.png")));
+      final ImageView pauseIcon = new ImageView(new Image(PlayAKMLTourSample.class.getResourceAsStream(
+          "/icons/pause.png")));
+
+      Button playPauseButton = new Button();
+      playPauseButton.setGraphic(playIcon);
+      playPauseButton.setDisable(true);
+
+      playPauseButton.setOnAction(e -> {
+        if (kmlTourController.getTour().getTourStatus() == KmlTourStatus.PLAYING) {
+          kmlTourController.pause();
+        } else {
+          kmlTourController.play();
+        }
+      });
+
+      // create replay button
+      final ImageView replayIcon = new ImageView(new Image(PlayAKMLTourSample.class.getResourceAsStream(
+          "/icons/replay.png")));
+
+      Button replayButton = new Button();
+      replayButton.setGraphic(replayIcon);
+      replayButton.setDisable(true);
+
+      replayButton.setOnAction(e -> {
+        kmlTourController.reset();
+        playPauseButton.setGraphic(playIcon);
+        playPauseButton.setDisable(false);
+      });
+
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setMaxSize(50, 100);
+      controlsVBox.getChildren().addAll(playPauseButton, replayButton);
+
+      // add a KML layer from a KML dataset with a KML tour
+      KmlDataset kmlDataset = new KmlDataset(new File("./samples-data/kml/Esri_tour.kmz").getAbsolutePath());
+      KmlLayer kmlLayer = new KmlLayer(kmlDataset);
+      scene.getOperationalLayers().add(kmlLayer);
+
+      kmlLayer.addDoneLoadingListener(() -> {
+        if (kmlLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // find the first KML tour in the dataset when loaded
+          KmlTour kmlTour = findFirstKMLTour(kmlDataset.getRootNodes());
+          if (kmlTour != null) {
+            // enable/disable buttons based on the KML tour status
+            kmlTour.addStatusChangedListener(kmlTourStatusChangedEvent -> {
+              switch (kmlTourStatusChangedEvent.getStatus()) {
+                case NOT_INITIALIZED:
+                  // before kml tour is added to controller
+                case INITIALIZING:
+                  // when kml tour is added to controller
+                  playPauseButton.setDisable(true);
+                  replayButton.setDisable(true);
+                  break;
+                case INITIALIZED:
+                  // when kml tour is ready to be played
+                  playPauseButton.setDisable(false);
+                  replayButton.setDisable(false);
+                  break;
+                case PAUSED:
+                  playPauseButton.setGraphic(playIcon);
+                  break;
+                case PLAYING:
+                  playPauseButton.setGraphic(pauseIcon);
+                  break;
+                case COMPLETED:
+                  playPauseButton.setDisable(true);
```

Comment:
when the KML tour has completed, it would be good to have a label alerting the user that it has done so.

Reply:
I thought the play/pause button being disabled might signal that.